internal/controllers/machinedeployment/mdutil/util_test.go

```diff
@@ -76,175 +76,150 @@ func generateDeployment(image string) clusterv1.MachineDeployment {
 			Annotations: make(map[string]string),
 		},
 		Spec: clusterv1.MachineDeploymentSpec{
-			Replicas: func(i int32) *int32 { return &i }(1),
+			Replicas: pointer.Int32(3),
 			Selector: metav1.LabelSelector{MatchLabels: machineLabels},
 			Template: clusterv1.MachineTemplateSpec{
 				ObjectMeta: clusterv1.ObjectMeta{
 					Labels: machineLabels,
 				},
-				Spec: clusterv1.MachineSpec{},
+				Spec: clusterv1.MachineSpec{
+					NodeDrainTimeout: &metav1.Duration{Duration: 10 * time.Second},
+				},
 			},
 		},
 	}
 }
 
-func generateMachineTemplateSpec(annotations, labels map[string]string) clusterv1.MachineTemplateSpec {
-	return clusterv1.MachineTemplateSpec{
+func TestEqualMachineTemplate(t *testing.T) {
+	machineTemplate := &clusterv1.MachineTemplateSpec{
 		ObjectMeta: clusterv1.ObjectMeta{
-			Annotations: annotations,
-			Labels:      labels,
+			Labels:      map[string]string{"l1": "v1"},
+			Annotations: map[string]string{"a1": "v1"},
+		},
+		Spec: clusterv1.MachineSpec{
+			NodeDrainTimeout:        &metav1.Duration{Duration: 10 * time.Second},
+			NodeDeletionTimeout:     &metav1.Duration{Duration: 10 * time.Second},
+			NodeVolumeDetachTimeout: &metav1.Duration{Duration: 10 * time.Second},
+			InfrastructureRef: corev1.ObjectReference{
+				Name:       "infra1",
+				Namespace:  "default",
+				Kind:       "InfrastructureMachine",
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+			},
+			Bootstrap: clusterv1.Bootstrap{
+				ConfigRef: &corev1.ObjectReference{
+					Name:       "bootstrap1",
+					Namespace:  "default",
+					Kind:       "BootstrapConfig",
+					APIVersion: "bootstrap.cluster.x-k8s.io/v1beta1",
+				},
+			},
 		},
-		Spec: clusterv1.MachineSpec{},
 	}
-}
 
-func TestEqualMachineTemplate(t *testing.T) {
+	machineTemplateWithEmptyLabels := machineTemplate.DeepCopy()
+	machineTemplateWithEmptyLabels.Labels = map[string]string{}
+
+	machineTemplateWithDifferentLabels := machineTemplate.DeepCopy()
+	machineTemplateWithDifferentLabels.Labels = map[string]string{"l2": "v2"}
+
+	machineTemplateWithEmptyAnnotations := machineTemplate.DeepCopy()
+	machineTemplateWithEmptyAnnotations.Annotations = map[string]string{}
+
+	machineTemplateWithDifferentAnnotations := machineTemplate.DeepCopy()
+	machineTemplateWithDifferentAnnotations.Annotations = map[string]string{"a2": "v2"}
+
+	machineTemplateWithDifferentNodeDrainTimeout := machineTemplate.DeepCopy()
+	machineTemplateWithDifferentNodeDrainTimeout.Spec.NodeDrainTimeout = &metav1.Duration{Duration: 20 * time.Second}
+
+	machineTemplateWithDifferentNodeDeletionTimeout := machineTemplate.DeepCopy()
+	machineTemplateWithDifferentNodeDeletionTimeout.Spec.NodeDeletionTimeout = &metav1.Duration{Duration: 20 * time.Second}
+
+	machineTemplateWithDifferentNodeVolumeDetachTimeout := machineTemplate.DeepCopy()
+	machineTemplateWithDifferentNodeVolumeDetachTimeout.Spec.NodeVolumeDetachTimeout = &metav1.Duration{Duration: 20 * time.Second}
```

Comment:
What about making a single machine template with all the inplace mutated fields changed?

Reply:
I think then we wouldn't actually test if a change in each of those fields results in an unequal MS (as the first difference already leads to that)